src/Native/Unix/System.Native/pal_process.c

```diff
@@ -233,15 +237,70 @@ int32_t SystemNative_ForkAndExecProcess(const char* filename,
     SystemNative_Pipe(waitForChildToExecPipe, PAL_O_CLOEXEC);
 #endif
 
-    // Fork the child process
-    if ((processId = fork()) == -1)
+    // The fork child must not be signalled until it calls exec(): our signal handlers do not
+    // handle being raised in the child process correctly
+    sigfillset(&signal_set);
+    pthread_sigmask(SIG_SETMASK, &signal_set, &old_signal_set);
```

Comment:
What happens if a signal like SIGCHLD is received between here and the later sigmask call?

Reply:
Then it gets delivered on the return from pthread_sigmask, but the only way for that to happen is an explicit kill.

Reply:
SIGCHLD will be handled by a different thread.